server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -160,16 +170,18 @@ public KnnSearchBuilder(
      * @param queryVector the query vector
      * @param k           the final number of nearest neighbors to return as top hits
      * @param numCands    the number of nearest neighbor candidates to consider per shard
+     * @param visitPercentage percentage of the total number of vectors to visit per shard
      */
     public KnnSearchBuilder(
         String field,
         VectorData queryVector,
         int k,
         int numCands,
+        float visitPercentage,
```

Comment:
if `visitPercentage` is optional, shouldn't we use `Float` instead, so null means that there is no value? It's similar to `similarity` (pun not intended).

That would make it more easy to reason with IMO

Reply:
agreed